DOCS-9659: Set due date rules (#26634)
DataDog/documentation: content/en/security/automation_pipelines/set_due_date.md

---
title: Set Due Date Rules
further_reading:
  - link: "/security/automation_pipelines"
    tag: "Documentation"
    text: "Automation Pipelines"
---

{{< callout url="https://www.datadoghq.com/product-preview/security-automation-pipelines/" >}}
  Automation Pipelines is in Preview. To enroll in the Preview for due date rules, click <strong>Request Access</strong>.
{{< /callout >}} 

Configure due date rules to ensure vulnerabilities are addressed within your specified SLO time frames. By setting these due dates, you can automate accountability, meet compliance requirements, and prioritize the prompt remediation of security issues, thereby preventing potential exploitation.

## Create a due date rule

1. On the [Automation Pipelines][2] page, click **Add a New Rule** and select **Set Due Date**.

> 1. Enter a descriptive name for the rule, for example, Cloud Infrastructure Anomaly Warnings.

Reviewer: Shouldn't there be quotes around "Cloud Infrastructure Anomaly Warnings"?

Author: If you're referring to style conventions, we don't have one for this. But I'd probably lean towards using bold vs. quotes for text that you would enter in a field. I can add that to the PR.

Reviewer: Definitely don't need it in quotes when adding it to the UI. Bold wuld be better.

---
title: Set Due Date Rules
further_reading:
  - link: "/security/automation_pipelines"
    tag: "Documentation"
    text: "Automation Pipelines"
---

{{< callout url="https://www.datadoghq.com/product-preview/security-automation-pipelines/" >}}
  Automation Pipelines is in Preview. To enroll in the Preview for due date rules, click <strong>Request Access</strong>.
{{< /callout >}} 

Configure due date rules to ensure vulnerabilities are addressed within your specified SLO time frames. By setting these due dates, you can automate accountability, meet compliance requirements, and prioritize the prompt remediation of security issues, thereby preventing potential exploitation.

## Create a due date rule

1. On the [Automation Pipelines][2] page, click **Add a New Rule** and select **Set Due Date**.
1. Enter a descriptive name for the rule, for example, **Cloud Infrastructure Anomaly Warnings**.
1. Use the following boxes to configure the rule criteria:

>     - **Any of these types**: The types of findings that the rule should check for. Available types include **Misconfiguration**, **Attack Path**, **Identity Risk**, and **API Security Finding**.

Author: Thanks. Do you know where I could see this in the UI? I'm not seeing it in any of the orgs right now.

>     - **Any of these types**: The types of findings that the rule should check for. Available types include **Misconfiguration**, **Attack Path**, **Identity Risk**, and **API Security Finding**.

Reviewer: <img width="1202" alt="Screenshot 2025-01-29 at 9 32 19 AM" src="https://github.com/user-attachments/assets/c404052f-2fc0-4aba-aadf-4e6b821a9ab1" />

https://dd.datad0g.com/security/configuration/pipeline-vulnerability/edit/6f0c2d16-9429-400a-a69a-0297e70c1014?ruleAction=set_due_date

>     - **Any of these types**: The types of findings that the rule should check for. Available types include **Misconfiguration**, **Attack Path**, **Identity Risk**, and **API Security Finding**.

Author: @JonRuben Thanks for the link.

P.S. I deleted your comment as this is a public repo, just to be safe.

    - **Any of these tags or attributes**: The resource tags or attributes that must match for the rule to apply.
1. Set a due date for each severity level, effective from the discovery of a matching severity vulnerability. Optionally, you can omit a due date if it is not necessary for a severity level. 
1. Click **Save**.

## Rule matching order

When Datadog identifies a vulnerability, it evaluates the vulnerability against your sequence of due date rules. Starting with the first rule, if there's a match, Datadog sets a due date on the vulnerability for the specified duration and stops evaluating further. If no match occurs, Datadog moves to the next rule. This process continues until a match is found or all rules are checked without a match.

## Removing due dates

When managing vulnerabilities, due dates can be removed under various conditions, such as:

- The detection rule that triggered the vulnerability passes successfully.
- The vulnerability is muted, either manually or automatically through a mute rule.
- The due date rule associated with the vulnerability is disabled or deleted.

> 

Reviewer 2: Also, the due date will be removed if the due date rule associated with the vulnerability has been modified and it doesn't match anymore the vulnerability.

Author: Ah, okay, that's good to know. I'll add that to the PR.

{{< /callout >}} 

Configure due date rules to ensure vulnerabilities are addressed within your specified SLO time frames. By setting these due dates, you can automate accountability, meet compliance requirements, and prioritize the prompt remediation of security issues, thereby preventing potential exploitation.

Notification rules are pre-set conditions that automate security issue notifications for your team. With notification rules, you don't need to manually set up alerts for each individual detection rule. Instead, you can configure rules to cover various scenarios by specifying factors like severity, rule type, and tags.

Notification rules help automatically alert your team about security problems. Instead of setting up alerts for each issue one by one, you use notification rules. You can set these rules to cover different situations by choosing options like severity, type, and specific tags.

## Create a due date rule

1. On the [Automation Pipelines][2] page, click **Add a New Rule** and select **Set Due Date**.
1. Enter a descriptive name for the rule, for example, **Cloud Infrastructure Anomaly Warnings**.
1. Use the following boxes to configure the rule criteria:

>     - **Any of these types**: The types of findings that the rule should check for. Available types include **Application Code Vulnerability**, **Application Library Vulnerability**, **Container Image Vulnerability**, **Misconfiguration**, **Attack Path**, **Identity Risk**, and **API Security Finding**.

Reviewer 3: I know this list is repeated in the other rule topics, so this isn't a big deal by any means if you don't want to hold up publication to get this change in all of them. I just find nested lists a lot easier to read than trying to visually parse non-bolded commas between bolded inline list items.
```suggestion
    - **Any of these types**: The types of findings that the rule should check for. Available types include:
      - **Application Code Vulnerability**
      - **Application Library Vulnerability**
      - **Container Image Vulnerability**
      - **Misconfiguration**
      - **Attack Path**
      - **Identity Risk**
      - **API Security Finding**
```

    - **Any of these tags or attributes**: The resource tags or attributes that must match for the rule to apply.

> 1. Set a due date for each severity level, effective from the discovery of a matching severity vulnerability. Optionally, you can omit a due date if it is not necessary for a severity level. 

Reviewer 3: ```suggestion
1. Set a due date for each severity level that requires one, effective from the discovery of a matching severity vulnerability.
```

1. Click **Save**.

## Rule matching order

When Datadog identifies a vulnerability, it evaluates the vulnerability against your sequence of due date rules. Starting with the first rule, if there's a match, Datadog sets a due date on the vulnerability for the specified duration and stops evaluating further. If no match occurs, Datadog moves to the next rule. This process continues until a match is found or all rules are checked without a match.

## Removing due dates

When managing vulnerabilities, due dates can be removed under various conditions, such as:

- The detection rule that triggered the vulnerability passes successfully.
- The vulnerability is muted, either manually or automatically through a mute rule.
- The due date rule associated with the vulnerability is disabled or deleted.
- The associated due date rule is modified so that its criteria no longer match the vulnerability.

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

> [2]: https://app.datadoghq.com/security/configuration/pipeline-vulnerability

Reviewer: Should we update this link to show due dates too?

Author: I think this link is part of Step 1 which tells users how to create a due date rule from the main Automation Pipelines page. So, it's intended to point to the main Automation Pipelines setup page.
